gno.land/pkg/sdk/vm/msgs.go

```diff
@@ -79,6 +88,59 @@ func (msg MsgAddPackage) GetReceived() std.Coins {
 	return msg.Deposit
 }
 
+//----------------------------------------
+// MsgEval
+
+// MsgEval - eval a Gno Expr.
+type MsgEval struct {
+	PkgPath string         `json:"pkg_path" yaml:"pkg_path"`
+	Expr    string         `json:"expr" yaml:"expr"`
+	Caller  crypto.Address `json:"caller" yaml:"caller"`
+
+	// XXX: This field is experimental, use with care as output is likely to change
+	// Default format is machine
+	Format Format `json:"format" yaml:"format"`
+}
+
+var _ std.Msg = MsgEval{}
+
+func NewMsgEval(format Format, pkgPath, expr string) MsgEval {
+	return MsgEval{
+		PkgPath: pkgPath,
+		Expr:    expr,
+		Format:  format,
+	}
+}
+
+// Implements Msg.
+func (msg MsgEval) Route() string { return RouterKey }
+
+// Implements Msg.
+func (msg MsgEval) Type() string { return "eval" }
+
+// Implements Msg.
+func (msg MsgEval) ValidateBasic() error {
+	if msg.Caller.IsZero() {
+		return std.ErrInvalidAddress("missing caller address")
+	}
+
+	if msg.Expr == "" {
+		return ErrInvalidExpr("missing expr to eval")
+	}
+
+	return nil
+}
+
+// Implements Msg.
+func (msg MsgEval) GetSignBytes() []byte {
+	return std.MustSortJSON(amino.MustMarshalJSON(msg))
```

Comment:
I support removing this and preventing MsgCall from being included in a transaction. It's just a query.